로그인 배경 추가

0214kbs opened the pull request:
### [관련 issue #]
#46 

### [리뷰 시 중점을 둘 곳]
- 로그인 배경 추가 : background 컴포넌트
- 로그인페이지에서 푸터 하단고정
- next-config 파일의 확장자를 mjs에서 js로 수정하였습니다. 

### [참고 사항(동작 화면 등)]
- 이후에 최적화할 예정입니다! 
![로그인배경 테스트 (1)](https://github.com/user-attachments/assets/56ccbd7e-fded-4cdb-87e0-8798e973b73c)

0214kbs committed:
conf: 영상 관련 패키지 설치 및 설정
- next-videos 설치

0214kbs committed:
feat: 로그인 배경 영상 추가
- 영상 직접 업로드
- autoplay, loop 설정

0214kbs committed:
feat: footer 하단 고정

0214kbs committed:
feat: 컴포넌트 높이 지정

0-Chan commented:
와우~ 적용 해주신 영상 보니까 영상은 깔끔하게 잘 적용 된 것 같습니다.

(최적화는 말씀하신 것 처럼 추후에 같이 고민 해보시죠! 예를 들면 login 페이지 자체에서 새로고침 시 제 화면에서는 반짝이는 현상이 있네요.)

0-Chan commented on src/app/(route)/login/page.tsx:
```diff
@@ -1,16 +1,15 @@
-      {/* <Video src={bgvideo} /> */}
-      <div className="flex flex-1 flex-col justify-center items-center">
+    <div>
+      <Background></Background>
```
해당 컴포넌트만 호출 하는거라면 `<Background/>`로만 써도 좋을 것 같아요!

0214kbs replied:
반영했습니다!

0-Chan commented on src/components/login/background/Background.tsx:
```diff
@@ -0,0 +1,32 @@
+    if (typeof window !== 'undefined') {
+      setHasWindow(true);
+    }
+  }, [hasWindow]);
```
useEffect 디펜던시에 hasWindow는 비워두어도 될 것 같습니다! 어짜피 마운트 시 한번만 호출 되는 구조 같아서요.

0214kbs replied:
맞네요 수정했습니다!

0-Chan commented on src/components/login/background/Background.tsx:
```diff
@@ -0,0 +1,32 @@
+            objectFit: 'cover',
+            position: 'fixed',
+          }}
+          src={require('../../../../public/login/loginbg.mp4')}
```
저희 mp4 파일 경로를 보니까 require 없이 경로를 쓰는건 어떨까요?

[Static assets](https://nextjs.org/docs/pages/building-your-application/optimizing/static-assets) 문서에 따르면
 `public/avatars/me.png` 경로에 있으면
  `/avatars/me.png`로 간단하게 적으면 된다고 합니다.

0214kbs replied:
넵 수정했습니다!

0-Chan commented on src/components/login/background/Background.tsx:
```diff
@@ -0,0 +1,32 @@
+    <>
+      {hasWindow && (
+        <video
+          autoPlay={true}
+          muted={true}
+          loop={true}
```
이건 머 필수 사항은 아닌데요ㅋㅋ

저는 기본 HTML 엘리먼트면 boolean 값은 굳이 안넣는게 간단해 보이더라구요. 

ex) `autoPlay={ture}` => `autoplay`

0214kbs replied:
맞네요 없애는 게 훨씬 깔끔하네요! 수정했습니다!

0214kbs committed:
refactor: 리뷰받은 부분 수정
- <Background></Background> 대신 <Background />로 수정
- useEffect 디펜던시에 hasWindow 제거
- 파일경로 require 제거
- 기본 hTML 엘리먼트의 boolean 값 제거

0214kbs committed:
fix: 외부 이미지 호출 오류 해결

0214kbs commented on next.config.js:
```diff
@@ -23,7 +25,5 @@ const nextConfig = {
 
-const withVideos = require('next-videos');
-module.exports = withVideos();
+module.exports = withVideos(nextConfig);
```
![image](https://github.com/user-attachments/assets/7fc7bcc2-9ad2-4f19-96e1-75dc131dd332)

위 이미지와 같이, https://lh3.googleusercontent.com에 호스팅된 이미지를 사용할 때 에러가 발생해서, next.config.js파일을 수정하였습니다.
